examples/vslm/main.go

```diff
@@ -0,0 +1,214 @@
+/*
+Copyright (c) 2017 VMware, Inc. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+/*
+This example program shows how the `view` and `property` packages can
+be used to navigate a vSphere inventory structure using govmomi.
+*/
+
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"github.com/vmware/govmomi/view"
+	"github.com/vmware/govmomi/vim25"
+	"github.com/vmware/govmomi/vim25/mo"
+	"github.com/vmware/govmomi/vslm/types"
+	"log"
+	"os"
+	"text/tabwriter"
+
+	"github.com/vmware/govmomi/examples"
+	vim "github.com/vmware/govmomi/vim25/types"
+	"github.com/vmware/govmomi/vslm"
+)
+
+func main() {
+	ctx := context.Background()
+
+	// Connect and login to vCenter (vpxd)
+	c, err := examples.NewClient(ctx)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	vslmClient, err := vslm.NewClient(ctx, c.Client)
+
+	if err != nil {
+		log.Fatal(err)
+	}
+	objectManager := vslm.NewGlobalObjectManager(vslmClient)
+	defer c.Logout(ctx)
+	createCmd := flag.NewFlagSet("create", flag.ExitOnError)
+	name := createCmd.String("name", "", "name for the created disk")
+	capacityInMB := createCmd.Int64("capacityInMB", 0, "capacity in MB of disk to create")
+	keepAfterDeleteVm := createCmd.Bool("keepAfterDeleteVm", true, "keep disk after VM it is attached to is deleted")
+	datastoreName := createCmd.String("datastoreName", "", "datastore name to create in")
+	datastoreMoRef := createCmd.String("datastoreMoRef", "", "datastore moref to create in")
+
+	listCmd := flag.NewFlagSet("list", flag.ExitOnError)
+
+	deleteCmd := flag.NewFlagSet("delete", flag.ExitOnError)
+	deleteID := deleteCmd.String("id", "", "id to delete")
+
+	if len(os.Args) < 2 {
+		fmt.Println("expected subcommand create, list, delete")
+		flag.Usage()
+		os.Exit(1)
+	}
+
+	switch (os.Args[1]) {
```

Comment:
Have you looked at the govc `disk.*` commands?  https://github.com/vmware/govmomi/blob/master/govc/USAGE.md#diskcreate

Sources are here: https://github.com/vmware/govmomi/tree/master/govc/disk

And tests: https://github.com/vmware/govmomi/blob/master/govc/test/disk.bats

Thinking we should adapt the disk commands to support using the global catalog, but that can be done later.

Reply:
Yeah, we can do that.  What's our backwards compatibility story?  The global object manager is only available from 6.7U2+

Reply:
In this case, when the `-ds` flag is given, the govc disk commands invoke the `ObjectManager` methods..  If no `-ds` flag is given, they invoke the `GlobalObjectManager` methods.  In the latter case, when run against VC < 6.7U2, the user just gets an error (e.g. MethodNotFound).  But again, we can look at doing this later.

Reply:
Realizing another issue with this example, CI fails because it runs all the examples against vcsim: https://github.com/vmware/govmomi/blob/66eb1fa1fa350800914d1a699fb66bea14099a28/govc/test/vcsim.bats#L20

But, vcsim doesn't support these new APIs yet.  For now we could exclude vslm from vcsim.bats or just not include the example for now.  I plan to add the vcsim support once your PR is merged.